### ISSUE TITLE
Translation function idea

### ISSUE THREAD

**Issue by rallek** [zikula/core#2820]
No idea if this is possible.

Today the translation workflow is hard. We can find a lot of bad translations because the translator do not know exactly where the string is used later.

Here my (crazy?) idea:
The normal translation string is given as following: `{gt text='My text for translation.'}`.  There is somewhere a function wich makes `My text for translation` if the string is not translated but the site is non English. Now I am looking for an admin setting `switch translation on/off`. If this is switched on a translation symbol is showing up behind the text. Clicking on this symbol a modal window appears. There we are able to translate the string. Closing the modal is saving the string in the *.po file. 

Somewhere in the admin area we can download the translations.

This functionality would make translating much more easy!

**Comment by Kaik** [comment]
I do not know if this is possible or at least it is not that easy to do * I think. What might be easy is to compare translation files in English *.po with other language *.po to look for empty 'strings'.

**Comment by rallek** [comment]
Thinking about that half a year later: The idea is still valid. Might be it can be a little bit more simple.

If the switch ``translation`` is ``on`` and the language of the site/browser is not english but the site do have multilanguage switched on a translation symbol should be behind the string (e.g. the german flag if I would browse my site from Germany). That should be possible. Of course this should be managed by permissions.

If you click on the symbol you do get the modal with the original string and the translation string from the po-file. Now you can fix a translation miskate or change what you want. And if the po file do not have this translation you can add it. 

This function would be really helpfull for translation teams!

**Comment by rallek** [comment]
Jenkins is a very good example how to proceed

**Comment by cmfcmf** [comment]
We already use the JMS Translation bundle and coud look into it's webinterface: http://jmsyst.com/bundles/JMSTranslationBundle/master/webinterface
Preview: http://vikbert.github.io/2016/01/14/How-to-i18n-in-Symfony2/jms_translation_gui.png

**Comment by Guite** [comment]
Blocked by https://github.com/schmittjoh/JMSTranslationBundle/issues/355 / https://github.com/schmittjoh/JMSTranslationBundle/pull/411

**Comment by Guite** [comment]
Closed as duplicate of #2425
